OathLab_Twitter/app/src/main/java/net/serkanbal/oathlab_twitter/MainActivity.java

```diff
@@ -0,0 +1,178 @@
+                    Log.d("Access_Token", "Your Access Token is: " + bearerToken);
+
+                    getTweets(bearerToken, mScreenName);
+
+                }
+            }
+        });
+    }
+
+    private void getTweets(String bearerToken, String screenName) {
+        OkHttpClient client = new OkHttpClient();
+
+        Request request = new Request.Builder()
+                .addHeader("Authorization", "Bearer " + bearerToken)
+                .url("https://api.twitter.com/1.1/statuses/user_timeline.json?screen_name=" +
+                        screenName + "&count=20")
+                .build();
+
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                e.printStackTrace();
+            }
+
+            @Override
+            public void onResponse(Call call, Response response) throws IOException {
+                if(!response.isSuccessful()) {
+                    throw new IOException("Problem at getting tweets" + response);
+                } else {
+                    Gson gson = new Gson();
+                    String str = response.body().string();
+                    TweetTextTimeStampObject[] array = gson.fromJson(str,
+                            TweetTextTimeStampObject[].class);
+
+                    mList.clear();
+                    for (int i = 0; i < array.length; i++) {
+                        String timestamp = array[i].getCreated_at();
+                        String text = array[i].getText();
+                        mList.add(new TwitterObject(timestamp, text));
+                    }
+
+//                        JSONArray rootArray = new JSONArray(response.body().string());
+//                        mList.clear();
+//                        for (int i = 0; i < rootArray.length(); i++) {
+//                            JSONObject obj = rootArray.getJSONObject(i);
+//                            String timestamp = obj.getString("created_at");
+//                            String text = obj.getString("text");
+//                            mList.add(new TwitterObject(timestamp, text));
+//                        }
+
+                    MainActivity.this.runOnUiThread(new Runnable() {
+                        @Override
+                        public void run() {
+                            RecyclerView recyclerview = (RecyclerView) findViewById(R.id.recyclerview);
+
+                            LinearLayoutManager linearLayoutManager = new LinearLayoutManager(MainActivity.this,
+                                    LinearLayoutManager.VERTICAL,false);
+
+                            recyclerview.setLayoutManager(linearLayoutManager);
+
+                            RecyclerView.Adapter adapter = new TwitterAdapter(mList);
+                            recyclerview.setAdapter(adapter);
+                            adapter.notifyDataSetChanged();
```

Comment:
You don't want to do all this work every time the user clicks the Search button. Set up the recycler view with the layout manager and adapter just once in `onCreate()`. Make the adapter a member variable, then here all you need to do is notify the adapter.